### PR TITLE
Refine cachedrequestloader tests

### PR DESCRIPTION
**Pull request by mTwTm**

### CODE REVIEW

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java [hunk 1]
```diff
@@ -29,6 +30,9 @@
     private int socketTag;
     private ResponseData stringLiveData;
     private static final String TAG = "CachedRequestLoader";
+    // Mainly for testing purposes.
+    private boolean delayCacheLoad = false;
```
Comment (cnevinc) [inline components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java]: can we add @VisibleForTesting for this?

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java [hunk 2]
```diff
@@ -50,6 +61,9 @@ public ResponseData getStringLiveData() {
     private void loadFromCache() {
         backgroundExecutorService.submit(() -> {
             try {
+                if (delayCacheLoad) {
+                    Thread.sleep(Inject.getTestDelay(delayCacheLoad));
```
Comment (cnevinc) [inline components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java]: can we  make this line like this:

`Inject.delayForTesting();`

and put `Thread.sleep(1000) `in Inject.java
```
void delayForTesting(
    Thread.sleep(1000);
}
```

so no one will accidentally run to Thread.sleep()... in production code

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/CachedRequestLoader.java [hunk 3]
```diff
@@ -41,16 +52,18 @@ public ResponseData getStringLiveData() {
...
+            } catch (ExecutionException | InterruptedException e) {
+                e.printStackTrace();
+                Log.e(TAG, "Failed to open Cache directory when reading cached banner config");
+            }
+        }, Inject.getTestDelay(delayCacheLoad));
     }
 
     private void loadFromRemote() {
-        new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag));
+        ThreadUtils.postToMainThreadDelayed(() -> new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag)), Inject.getTestDelay(delayNetworkLoad));
```
Comment (cnevinc) [inline components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/CachedRequestLoader.java]: This is the same. Maybe use Injection to decide if we want to delay here ?
e.g. in Inject.java, under main:

```
void loadFromRemoteImp(){
    new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag));
}
```

and under test will be 
```
void loadFromRemoteImp(){
    ThreadUtils.postToMainThreadDelayed(() -> new RemoteLoadUrlTask(stringLiveData, this).execute(subscriptionUrl, userAgent, Integer.toString(socketTag)), Inject.getTestDelay(delayNetworkLoad));
}
```

File: components/utils/cachedrequestloader/src/androidTest/java/org/mozilla/cachedrequestloader/CachedRequestLoaderTest.java [hunk 4]
```diff
@@ -45,33 +47,76 @@ public void testLoadAndCache() throws InterruptedException {
...
                 Assert.assertEquals(RESPONSE_BODY2, s.second);
+                latch2.countDown();
+            }
+        };
+        CachedRequestLoader cachedRequestLoader = new CachedRequestLoader(InstrumentationRegistry.getContext(), KEY, targetUrl, null, SOCKET_TAG, false, true);
+        LiveData<Pair<Integer, String>> liveData = cachedRequestLoader.getStringLiveData();
+        liveData.observeForever(observer);
+        latch1.await();
+        // Wait one second so the write to cache finishes.
+        Thread.sleep(1000);
```
Comment (cnevinc) [inline components/utils/cachedrequestloader/src/androidTest/java/org/mozilla/cachedrequestloader/CachedRequestLoaderTest.java]: ditto

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java (hunk 1 above)
Comment (mTwTm) [reply components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java]: They are already private. I'm not sure if that makes sense.

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java (hunk 2 above)
Comment (mTwTm) [reply components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java]: Fixed.

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/CachedRequestLoader.java (hunk 3 above)
Comment (mTwTm) [reply components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/CachedRequestLoader.java]: Fixed.

File: components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java (hunk 1 above)
Comment (mTwTm) [reply components/utils/cachedrequestloader/src/main/java/org/mozilla/cachedrequestloader/BackgroundCachedRequestLoader.java]: Fixed.

**Pull request merged**